pandas/_libs/lib.pyx

```diff
@@ -67,7 +67,8 @@ def memory_usage_of_objects(ndarray[object, ndim=1] arr):
     cdef int64_t s = 0
 
     n = len(arr)
-    for i from 0 <= i < n:
+    # Hrm why was this 0 <= i < n
```

Comment:
why do you think this is the issue?

Reply:
This is ok, I think are definition of memory_usage should be slightly updated. (in core/base.py) as its meant for pure object dtypes, not sparse.

Reply:
Should it be updated?

My understanding is that memory_usage for a SparseSeries should equal to the memory_usage of the index plus the memory_usage of the SparseArray

The original bug is honestly an edgecase because if you have more than one dense element it doesn't fail:

```python
import pandas as pd
ss = pd.SparseSeries(['a', 'b', 'c'], fill_value='a')

ss.memory_usage(deep=True) #=> No segfault because the dense elements is len > 0

ss= pd.SparseSeries(['a', 'a', 'a'], fill_value='a')
ss.memory_usage(deep=True) #=> Segfault because dense elements len = 0
```

I'm not sure core/base.py should be updated. The problem with doing that would be that I would do a length check before handing it off to the lower level pyx file. This fix gets rid of the possible segfault by accessing memory that's not there.

Reply:
Also I meant to take out my thought process comment ;). I'll do that now